data-prepper-plugins/parquet-codecs/src/test/java/org/opensearch/dataprepper/plugins/codec/parquet/ParquetInputCodecTest.java

```diff
@@ -0,0 +1,201 @@
+        FileSystem fs = FileSystem.get(new Configuration());
+        String filepath = path.toUri().getPath();
+        fs.delete(new Path(filepath));
+        fs.close();
+    }
+
+    private static InputStream createRandomParquetStream(int numberOfRecords) throws IOException {
+
+        Schema schema = parseSchema();
+        String OS = System.getProperty("os.name").toLowerCase();
+
+        if (OS.contains("win")) {
+            System.setProperty("hadoop.home.dir", Paths.get("").toAbsolutePath().toString());
+        } else {
+            System.setProperty("hadoop.home.dir", "/");
+        }
+
+        List<GenericData.Record> recordList = generateRecords(schema, numberOfRecords);
+
+        path = Paths.get("test-parquet.parquet");
+
+        try(ParquetWriter<GenericData.Record> writer = AvroParquetWriter.<GenericData.Record>builder(new Path(path.toUri().getPath()))
+                .withSchema(schema)
+                .withCompressionCodec(CompressionCodecName.SNAPPY)
+                .withRowGroupSize(ParquetWriter.DEFAULT_BLOCK_SIZE)
+                .withPageSize(ParquetWriter.DEFAULT_PAGE_SIZE)
+                .withConf(new Configuration())
+                .withValidation(false)
+                .withDictionaryEncoding(false)
+                .build()) {
+            for (GenericData.Record record : recordList) {
+                writer.write(record);
+            }
+        }
+        fileInputStream = new FileInputStream(path.toString());
+        return fileInputStream;
+    }
+
+    private static Schema parseSchema() {
+        String schemaJson = "{\"namespace\": \"org.myorganization.mynamespace\","
+                + "\"type\": \"record\","
+                + "\"name\": \"myrecordname\","
+                + "\"fields\": ["
+                + " {\"name\": \"myString\",  \"type\": [\"string\", \"null\"]}"
+                + ", {\"name\": \"myInteger\", \"type\": \"int\"}"
+                + " ]}";
+
+        Schema.Parser parser = new Schema.Parser().setValidate(true);
+        return parser.parse(schemaJson);
+    }
+
+    private static List<GenericData.Record> generateRecords(Schema schema, int numberOfRecords) {
+
+        List<GenericData.Record> recordList = new ArrayList<>();
+
+        for(int recordValue = 1; recordValue <= numberOfRecords; recordValue++) {
+            GenericData.Record record = new GenericData.Record(schema);
+            record.put("myInteger", recordValue);
+            record.put("myString", recordValue + RandomString.make());
+            recordList.add(record);
+        }
+        return recordList;
+    }
+
+    private static InputStream createInvalidParquetStream() {
+        String OS = System.getProperty("os.name").toLowerCase();
+        if (OS.contains("win")) {
+            System.setProperty("hadoop.home.dir", Paths.get("").toAbsolutePath().toString());
+        }
+        return  new ByteArrayInputStream(INVALID_PARQUET_INPUT_STREAM.getBytes());
+    }
+}
```

Comment:
After adding metrics in the exception cases (as commented above), add test cases to test the metrics.